src/cpp/transport/stream_framing/StreamFramingProtocol.cpp

```diff
@@ -255,7 +274,7 @@ size_t FramingIO::read_framed_msg(
                         {
                             state_ = InputState::UXR_FRAMING_READING_SRC_ADDR;
                         }
-                        else if (0 < transport_read(timeout, transport_rc))
+                        else if (0 < transport_read(timeout, transport_rc, msg_len_ - msg_pos_))
```

Comment:
```suggestion
                        else if (0 < transport_read(timeout, transport_rc, (msg_len_ - msg_pos_) + 2))
```
I think we should include the CRC read here

Reply:
Done https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/267/commits/937ea81d38832563e03a6d95c6986d322bc5b8b1